Sources/SwiftDocC/Infrastructure/Diagnostics/Problem.swift

```diff
@@ -21,6 +21,10 @@ public struct Problem {
         self.diagnostic = diagnostic
         self.possibleSolutions = Array(possibleSolutions)
     }
+
+    public init(diagnostic: Diagnostic) {
```

Comment:
What are your thoughts on offering this "convenience" initializer? I feel a lot of checkers initialize a problem with just a `diagnostic` and no possible solutions.

To keep the scope of this PR contained, I can create a second PR when this one is merged to update all the other checkers to use this initializer when applicable.

Reply:
Doing a project-wide search for `Problem(diagnostic: diagnostic, possibleSolutions: []),` there are 54 instances of this.

Reply:
I’m fine either way with that convenience initializer.

You’re right that there are some ergonomics issues with creating diagnostics. I feel that that is having some impact on notes and solutions. A number problems that are diagnosing unexpected extra content could offer to remove that and some problems that are diagnosing missing required information could offer to insert templated content to be filled out.

A targeted effort could be made to update each diagnostic but it would also be a good idea to review the API for creating diagnostics—both of which seem outside the scope of SR-15316.